### PR TITLE
Profile

### PR DESCRIPTION
**Pull request by AlirieGray**
* Added view and edit profile components
* Fixed a few minor style issues such as shrinking the size of the nav pills in the rubric show view
* Added icons to nav dropdown
* General cleanup including removing commented out code and warnings

### CODE REVIEW

File: src/components/EditProfile.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            <div className="card-content">
+              <h4 className="card-title">Edit Profile  -
+                <small className="category">  Complete your profile</small>
+              </h4>
+              <form>
+                <div className="row">
+                  <div className="col-md-3">
+                    <div className="form-group label-floating is-empty">
+                      <label className="control-label">Username</label>
+                      <input type="text" className="form-control" value={this.state.username} onChange={(e) => this.setState({username: e.target.value})}/>
```
Comment (tlambrou) [inline src/components/EditProfile.js]: Why not deconstruct all the state variables above the return?...
`const { username, email, firstName, lastName, bio, userId } = this.state`

File: src/components/EditProfile.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                    <span className="material-input"></span></div>
+                  </div>
+                  <div className="col-md-4">
+                    <div className="form-group label-floating is-empty">
+                      <label className="control-label">Email address</label>
+                      <input type="email" className="form-control" value={this.state.email} onChange={(e) => this.setState({email: e.target.value})}/>
```
Comment (tlambrou) [inline src/components/EditProfile.js]: Why not deconstruct all the state variables above the return?...
`const { username, email, firstName, lastName, bio, userId } = this.state`

File: src/components/EditProfile.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                    <span className="material-input"></span></div>
+                  </div>
+                  </div>
+                  <div className="row">
+                    <div className="col-md-6">
+                      <div className="form-group label-floating is-empty">
+                        <label className="control-label">First Name</label>
+                        <input type="text" className="form-control" value={this.state.firstName} onChange={(e) => this.setState({firstName: e.target.value})}/>
```
Comment (tlambrou) [inline src/components/EditProfile.js]: Why not deconstruct all the state variables above the return?...
`const { username, email, firstName, lastName, bio, userId } = this.state`

File: src/components/EditProfile.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                      <span className="material-input"></span></div>
+                    </div>
+                    <div className="col-md-6">
+                      <div className="form-group label-floating is-empty">
+                        <label className="control-label">Last Name</label>
+                        <input type="text" className="form-control" value={this.state.lastName} onChange={(e) => this.setState({lastName: e.target.value})}/>
```
Comment (tlambrou) [inline src/components/EditProfile.js]: Why not deconstruct all the state variables above the return?...
`const { username, email, firstName, lastName, bio, userId } = this.state`

File: src/components/EditProfile.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+                      <span className="material-input"></span></div>
+                    </div>
+                  </div>
+                  <div className="row">
+                    <div className="col-md-12">
+                      <div className="form-group">
+                        <label>About Me</label>
+                        <div className="form-group label-floating is-empty">
+                          <textarea className="form-control" rows="5" value={this.state.bio} onChange={(e) => this.setState({bio: e.target.value})} ></textarea>
```
Comment (tlambrou) [inline src/components/EditProfile.js]: Why not deconstruct all the state variables above the return?...
`const { username, email, firstName, lastName, bio, userId } = this.state`

File: src/components/EditProfile.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+              </form>
+            </div>
+          </div>
+        </div>
+    </div>
+    );
+  }
+}
+
+export default EditProfile;
```
Comment (tlambrou) [inline src/components/EditProfile.js]: Remember that you need to wrap components that need to be rendered in the react-router pipeline with `withRouter()` even if you're not mapping to props etc..

File: src/components/EditProfile.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+
+class EditProfile extends Component {
```
Comment (tlambrou) [inline src/components/EditProfile.js]: Suggestion: We could just make one component that is both show and edit view for the user's profile.  Then we can just pass in a different user and reuse the same component if we want to allow the user to see other people's profile pages. :)

File: src/components/NavTop.js [hunk 8]
```diff
@@ -47,119 +48,35 @@ class NavTop extends Component {
...
-                        </a>
-                      </li>
-                      <li>
-                        <a href="">
-                          You have 5 new tasks
-                        </a>
-                      </li>
-                      <li>
-                        <a href="">
-                          You're now friend with Andrew
```
Comment (tlambrou) [inline src/components/NavTop.js]: The apostrophe might break a linter.  We can use the entity `&apos;` instead.

Comment (tlambrou) [reply src/components/NavTop.js]: Nevermind.  You deleted this

File: src/components/Profile.js [hunk 9]
```diff
@@ -1,16 +1,41 @@
 import React, { Component } from 'react';
 
 class Profile extends Component {
-
   render() {
+    const user = {
```
Comment (tlambrou) [inline src/components/Profile.js]: I know I wrote this code before, but we could just get rid of the user object and just make the variables first level.

File: src/components/Profile.js [hunk 10]
```diff
@@ -1,16 +1,41 @@
...
+              <h4 className="card-title"> My Profile </h4>
+              <hr />
+              <div col-sm-6>
+                <p> Username: <small> {user.username} </small> </p>
+              </div>
+              <div col-sm-6>
+                <p> Email: <small> {user.email} </small> </p>
+              </div>
+              <div col-sm-6>
+                <p> First Name: <small> {user.firstName !== 'null' ? user.firstName : <a className="text-info" href={`/${user.username}/edit`}> Click here to add a first name </a>} </small> </p>
```
Comment (tlambrou) [inline src/components/Profile.js]: Noice

File: src/components/Routes.js [hunk 11]
```diff
@@ -23,7 +22,7 @@ class Routes extends Component {
         return <Redirect from='/' to='/dashboard' exact />
       }
       return <Sidebar />
-    } else if (this.props.location.pathname === '/') {
+    } else if (this.props.location.pathname === '/' || this.props.location.pathname === '/dashboard') {
```
Comment (tlambrou) [inline src/components/Routes.js]: Schweet!

File: src/components/ScoreCard.js [hunk 12]
```diff
@@ -30,8 +30,9 @@ const CompetencyScore = withRouter(
       const totalSum = Competencies.reduce((acc, currentComp, currentCompIndex, competencies) => {
         return acc + (this.calculateCompetencyScore(currentComp)/ 100)
       }, 0)
-      const result = (totalSum/Competencies.length) * 100
-      return result.toFixed(0)
+      const result = (totalSum/Competencies.length) * 100;
+
+      return isNaN(result) ? 0 : result.toFixed(0)
```
Comment (tlambrou) [inline src/components/ScoreCard.js]: Yay!

Comment (tlambrou) [reply src/components/ScoreCard.js]: Is that a native function or did you write it?

File: src/components/Sidebar.js [hunk 13]
```diff
@@ -52,33 +57,35 @@ class Sidebar extends Component {
...
+                      </Link>
+                    </li>
+                    <li>
+                      <Link to={`/${user.username}/edit`}>
+                        <span className="sidebar-mini"> <i className="material-icons">settings</i> </span>
+                        <span className="sidebar-normal"> Edit Profile </span>
+                      </Link>
+                    </li>
+                    <li>
+                      <a href="#" onClick={() => {
```
Comment (tlambrou) [inline src/components/Sidebar.js]: I think the fact that this is both a link and has an onclick is not a good choice.  We need to stick with the onclick and remove the link.  Make a span or div.

File: src/components/ScoreCard.js (hunk 12 above)
Comment (AlirieGray) [reply src/components/ScoreCard.js]: native function!

File: src/components/Sidebar.js (hunk 13 above)
Comment (AlirieGray) [reply src/components/Sidebar.js]: Yeah I know but when I changed it, it messed up the style :/

File: src/components/EditProfile.js (hunk 5 above)
Comment (AlirieGray) [reply src/components/EditProfile.js]: will do

File: src/components/EditProfile.js (hunk 6 above)
Comment (AlirieGray) [reply src/components/EditProfile.js]: will do

File: src/components/NavTop.js (hunk 8 above)
Comment (AlirieGray) [reply src/components/NavTop.js]: Yeah it wasn't being used, and we can always add it later if we want it back

File: src/components/EditProfile.js (hunk 5 above)
Comment (Jeffchiucp) [reply src/components/EditProfile.js]: good job

**Pull request merged**